### PR TITLE
Remove references to settings.xml

### PR DESCRIPTION
**Pull request by MatthewKhouzam**
There is none

### CODE REVIEW

File: .github/workflows/maven-publish.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
           server-password: MAVEN_PASSWORD
           gpg-passphrase: PASSPHRASE
       - name: Make a snapshot
-        run: mvn -Pdeploy -Pproduction --no-transfer-progress --batch-mode clean deploy -U
+        run: mvn --no-transfer-progress --batch-mode clean deploy -U
```
Comment (bhufmann) [inline .github/workflows/maven-publish.yml:35]: the profiles should stay (-Pdeploy and -Pproduction)

File: .github/workflows/maven-publish.yml [hunk 2]
```diff
@@ -44,14 +44,13 @@ jobs:
...
-          server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
-          settings-path: ${{ github.workspace }} # location for the settings.xml file
+          server-id: ossrh
           server-username: MAVEN_USERNAME
           server-password: MAVEN_PASSWORD
           gpg-passphrase: PASSPHRASE
 
       - name: Publish to Maven Central Repository
-        run: mvn --batch-mode deploy -s $GITHUB_WORKSPACE/settings.xml
+        run: mvn --batch-mode deploy
```
Comment (bhufmann) [inline .github/workflows/maven-publish.yml:53]: shouldn't it also have the profiles here?

**Review by bhufmann — CHANGES_REQUESTED** [review]

File: .github/workflows/maven-publish.yml [hunk 3]
```diff
@@ -44,14 +44,13 @@ jobs:
           java-version: '11'
           distribution: 'temurin'
           cache: 'maven'
-          server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
```
Comment (marcdumais-work) [inline .github/workflows/maven-publish.yml:47]: In the GitHub documentation example, the `server-id` line is kept:
https://docs.github.com/en/actions/publishing-packages/publishing-java-packages-with-maven#publishing-packages-to-the-maven-central-repository-and-github-packages

update: seems the default value is "github", so you want to keep the line:
https://github.com/actions/setup-java#maven-options

**Comment by marcdumais-work** [comment]
An advanced example from the `setup-java` repo : close to our use-case but there are small differences that you could look-into if this still does not work after responding to current comments:
https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#publishing-using-apache-maven

File: .github/workflows/maven-publish.yml (hunk 3 above)
Comment (marcdumais-work) [reply .github/workflows/maven-publish.yml:47]: In the GH doc above, they provide some corresponding content of the generated settings.xml - you should "cat" the one(s) generated in your workflow and compare, making sure the parameters you pass tp `setup-java` are reflected in there.

Comment (MatthewKhouzam) [reply .github/workflows/maven-publish.yml:47]: `ossrh` is for maven central AFAIK
`github` is for maven central

File: .github/workflows/maven-publish.yml (hunk 1 above)
Comment (MatthewKhouzam) [reply .github/workflows/maven-publish.yml:35]: Ok!

**Review by bhufmann — APPROVED** [review]

**Pull request merged**